src/ceph.in

```diff
@@ -899,7 +899,10 @@ def main():
     try:
         if childargs and childargs[0] == 'ping':
             return ping_monitor(cluster_handle, childargs[1], timeout)
-        run_in_thread(cluster_handle.connect, timeout=timeout)
+        result = run_in_thread(cluster_handle.connect, timeout=timeout)
+        if type(result) is tuple and result[0] == -errno.EINTR:
```

Comment:
I agree that the return value from run_in_thread is strange.

I'm not aware of the revision history here, but I think KeyboardInterrupt is very unlikely to be caught there since it's caught (and, strangely, raised so as to be caught) by run_in_thread (and ping_monitor primarily uses run_in_thread as well).

In fact, I think this issue is now more pervasive; squashing the KeyboardInterrupt leads to an infinite loop here and I suspect it may lead to bad behavior at other unchecked call sites.